resolwe/flow/tests/test_test_framework.py

```diff
@@ -49,3 +56,94 @@ def date_in_line(x):
+        def test_assert_json_storage_id(self, get_mock):
+            example_json = {'foo': [1.0, 2.5, 3.14], 'bar': ['ba', 'cd']}
+
+            dummy_case = ProcessTestCase.__new__(ProcessTestCase)
+            dummy_case.files_path = ''
+            dummy_case._debug_info = lambda _: ''
+            dummy_case.assertEqual = self.assertEqual
+
+            obj_mock = mock.MagicMock()
+
+            storage_id = 'no_id'
+            storage_mock = mock.MagicMock(spec=Storage)
+            storage_mock.id = storage_id
+            storage_mock.json = example_json
+            get_mock.side_effect = [storage_mock]
+
+            # use in-memory binary stream object for speed and simplicity
+            gzipped_json_file = io.BytesIO()
+            with gzip.open(gzipped_json_file, mode='wt') as f:
+                json.dump(example_json, f)
+            # set seek position of the binary stream object back to 0
+            gzipped_json_file.seek(0)
+
+            join_mock = mock.MagicMock(side_effect=[gzipped_json_file])
+            isfile_mock = mock.MagicMock(return_value=True)
+            with mock.patch.object(os.path, 'join', join_mock):
+                with mock.patch.object(os.path, 'isfile', isfile_mock):
+                    ProcessTestCase.assertJSON(dummy_case, obj_mock, storage_id, '', 'foo.gz')
+
+        @unittest.skipIf(sys.version_info.major < 3, "Needs Python 3 version of the gzip module")
+        def test_assert_json_file_missing(self):
+            example_json = {'foo': [1.0, 2.5, 3.14], 'bar': ['ba', 'cd']}
+
+            dummy_case = ProcessTestCase.__new__(ProcessTestCase)
+            dummy_case.files_path = ''
+            dummy_case._debug_info = lambda _: ''
+            dummy_case.assertEqual = self.assertEqual
+
+            obj_mock = mock.MagicMock()
+
+            storage_mock = mock.MagicMock(spec=Storage)
+            storage_mock.id = 'no_id'
+            storage_mock.json = example_json
+
+            # use in-memory binary stream object for speed and simplicity
+            gzipped_json_file = io.BytesIO()
+            with gzip.open(gzipped_json_file, mode='wt') as f:
+                json.dump(example_json, f)
+
+            # another binary stream object to test automatic creation of gzipped json files
+            new_gzipped_json_file = io.BytesIO()
+
+            join_mock = mock.MagicMock(side_effect=[new_gzipped_json_file])
+            isfile_mock = mock.MagicMock(return_value=False)
+            with mock.patch.object(os.path, 'join', join_mock):
+                with mock.patch.object(os.path, 'isfile', isfile_mock):
+                    with six.assertRaisesRegex(self, AssertionError,
+                                               'Output file .* missing so it was created.'):
+                        ProcessTestCase.assertJSON(dummy_case, obj_mock, storage_mock, '', 'foo.gz')
+            self.assertEqual(gzipped_json_file.getvalue(), new_gzipped_json_file.getvalue())
```

Comment:
@dblenkus, thanks for your comment. I've changed the test to compare the unzipped JSON with `example_json`. Please, re-review!